src/v2/layouts/MemberRegistration/MemberRegistrationAPIComponent.tsx

```diff
@@ -0,0 +1,34 @@
+import { useAPI } from '../../components/hooks';
+import { useEffect } from 'react';
+import {
+	memberRegistrationAPI,
+	MemberRegistrationPayload,
+	MemberRegistrationResponseProps,
+} from './memberRegistrationAPI';
+
+export type MemberAPIProps = {
+	data: MemberRegistrationPayload;
+	submitCallback: (resp: MemberRegistrationResponseProps) => void;
+};
+export const MemberAPI = ({ data, submitCallback }: MemberAPIProps): JSX.Element => {
+	const { loading, error, response } = useAPI(
+		memberRegistrationAPI(data as MemberRegistrationPayload)
+	);
+	useEffect(() => {
+		if (response || error) {
+			submitCallback({ response, error });
```

Comment:
Handle the loading, response and error in parent component.
```suggestion
			submitCallback({ loading, response, error });
```